internal/repository/repository.go

```diff
@@ -252,12 +322,51 @@ func (r *Repository) LookupBlobSize(id restic.ID, tpe restic.BlobType) (uint, bo
 	return r.idx.LookupSize(restic.BlobHandle{ID: id, Type: tpe})
 }
 
+func (r *Repository) getZstdEncoder() *zstd.Encoder {
+	r.allocEnc.Do(func() {
+		level := zstd.SpeedDefault
+		if r.opts.Compression == CompressionMax {
+			level = zstd.SpeedBestCompression
+		}
+
+		enc, err := zstd.NewWriter(nil, zstd.WithEncoderLevel(level))
+		if err != nil {
+			panic(err)
+		}
+		r.enc = enc
+	})
+	return r.enc
+}
+
+func (r *Repository) getZstdDecoder() *zstd.Decoder {
+	r.allocDec.Do(func() {
+		dec, err := zstd.NewReader(nil)
```

Comment:
```suggestion
		dec, err := zstd.NewReader(nil, zstd.WithDecoderConcurrency(0))
```
... by default 4 decoders are allocated. Might as well have GOMAXPROCS.